### PR TITLE
Add generic contracts for object authenticity management

### PR DESCRIPTION
**Pull request by jnmt**
## Description

This PR adds a part of the generic contracts. We will provide generic contracts and functions so that users do not have to write their own contracts and functions for common use cases.

Generic contracts provide two main functionalities: object authenticity management and collection authenticity management. This PR only adds the former as the first step. In object authenticity management, users can check the authenticity of an object by storing the hash value of the object in ScalarDL and comparing it later with the one calculated based on the current object.

See the design doc for the detailed specifications. Sorry most parts except for input/output and examples are currently Japanese. If you have any questions, please feel free to ask.
https://docs.google.com/document/d/1hSldq-D5mSreV4DK7OZOA7XUXTd71MWXPzoQ3iUwoow/edit?usp=sharing

## Related issues and/or PRs

N/A

## Changes made

- Add generic contracts for object authenticity management
- Add basic CI settings that run unit tests, spotless, and spotbugs

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes. **Will be added after all-related PRs are merged**
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

The following will be added by separate PRs each.
- Other generic contracts for collection management
- A generic function for consistent state management of the objects
- Integration tests or end-to-end tests (to be discussed)

## Release notes

Added generic contracts for object authenticity management.

### CODE REVIEW

File: generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  public static final String PROPERTIES = "properties";
+  public static final String STATUS = "status";
+  public static final String FAULTY_VERSIONS = "faulty_versions";
+
+  public static final String STATUS_NORMAL = "normal";
+  public static final String STATUS_FAULTY = "faulty";
+
+  // Error messages
+  public static final String OBJECT_ID_IS_MISSING_OR_INVALID =
+      "The object ID is not specified in the arguments or invalid.";
```
Comment (josh-wong) [inline generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java:17]:
```suggestion
      "The object ID is not specified in the arguments or is invalid.";
```

File: generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  public static final String HASH_VALUE_IS_MISSING_OR_INVALID =
+      "The hash value is not specified in the arguments or invalid.";
```
Comment (josh-wong) [inline generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java:19]:
```suggestion
      "The hash value is not specified in the arguments or is invalid.";
```

File: generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  public static final String HASH_VALUES_ARE_MISSING =
+      "The hash values are not specified in the arguments.";
+  public static final String INVALID_PROPERTIES_FORMAT =
+      "The specified format of the properties is not valid.";
```
Comment (josh-wong) [inline generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java:23]: For consistency with the other messages above, how about if we change this to `invalid`?

```suggestion
      "The specified format of the properties is invalid.";
```

File: generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  public static final String INVALID_HASH_VALUES_FORMAT =
+      "The specified format of the hash values is not valid.";
```
Comment (josh-wong) [inline generic-contracts-and-functions/src/main/java/com/scalar/dl/client/Constants.java:25]: For consistency with the other messages above, how about if we change this to `invalid`?

```suggestion
      "The specified format of the hash values is invalid.";
```

**Review by josh-wong — CHANGES_REQUESTED** [review]
I've added some minor suggestions. PTAL!

**Review by supl — APPROVED** [review]
LGTM!
Thank you.
I think it's nice to have them.

**Comment by jnmt** [comment]
As discussed with Hiro, we decided to include generic contracts and functions in the scalar-labs/scalar main repository so that we can easily run e2e tests with generic contracts. So, I will close this and re-create a PR in it.

@josh-wong @supl Thank you for the review and suggestions. I will apply them in the re-created PR, so PTAL it again. Sorry for the inconvenience.

**Pull request closed without merging**